### PR TITLE
Fixes #344: Direct relationship jest matcher

### LINKED ISSUE #344: Add `toMatchRelationshipSchema()` matcher
The `.toMatchRelationshipSchema()` matcher should ensure that all properties are non-arrays or objects.

### PR DESCRIPTION
**Pull request by ndowmon**

### CODE REVIEW

File: packages/integration-sdk-testing/src/jest.ts [hunk 1]
```diff
@@ -222,12 +226,68 @@ export function toMatchGraphObjectSchema<T extends Entity>(
...
+      {
+        $ref: '#GraphObject',
+      },
+      {
+        additionalProperties: {
+          type: ['boolean', 'integer', 'null', 'number', 'string'],
+        },
+      },
+    ],
+  };
```
Comment (ndowmon) [inline packages/integration-sdk-testing/src/jest.ts:267]: There is no `#Relationship` schema in the data model, but this defines what it _would_ look like.

The tricky part will be figuring how to share logic between `ExplicitRelationship` and `MappedRelationship`. Not sure if we will need one JSON schema for each or what.

File: packages/integration-sdk-testing/src/jest.ts [hunk 2]
```diff
@@ -238,12 +298,38 @@ export function toMatchGraphObjectSchema<T extends Entity>(
...
+    return {
+      message: () => 'Success!',
+      pass: true,
+    };
+  } else {
+    return hasDistinctKeysResult;
+  }
+}
+
+function toHaveDistinctKeys(received: { _key: string }[]) {
```
Comment (austinkelleher) [inline packages/integration-sdk-testing/src/jest.ts:313]: This is minor, but I think we could improve this function by not returning a the jest matcher result format and instead return only a boolean. That will make the function more pure and the caller can decide what to do with the resulting boolean. Also I think we can improve the TS typings. Something like the following (untested):

```ts
function graphObjectHasDistinctKeys<
  T extends Entity | ExplicitRelationship
>(received: T[]): boolean {
  const _keys = received.map((r) => r._key);
  return Array.isArray(_keys) && new Set(_keys).size === _keys.length;
}
```

Caller:

```ts
if (graphObjectHasDistinctKeys(received)) {
  return {
    message: () => 'Success!',
    pass: true,
  };
} else {
  return {
    message: () =>
      `Object \`_key\` properties array is not unique: [${_keys}]`,
    pass: false,
  };
}
```

**Review by austinkelleher — APPROVED** [review]
This is awesome! Thanks a lot for taking this on. I left a minor code improvement comment, but approving anyway.

File: packages/integration-sdk-testing/src/jest.ts (hunk 2 above)
Comment (ndowmon) [reply packages/integration-sdk-testing/src/jest.ts:313]: Good point. I had originally considered exposing the `toHaveDistinctKeys()` matcher from the SDK but now I'm not sure that makes sense.

Comment (ndowmon) [reply packages/integration-sdk-testing/src/jest.ts:313]: I'll have to rebase this back off of master (it's branched off of `add-dotenv-config-to-jest-config` which it seems like I'll close without merging). I think it's a good suggestion and worth doing while I'm doing the rebase.

**Review by austinkelleher — APPROVED** [review]
Excellent!

**Comment by aiwilliams** [comment]
Related to #228

**Pull request merged**